src/fw-update/fw-update-device.cpp

```diff
@@ -210,8 +213,14 @@ namespace librealsense
         // WaitForDFU state sends several DFU_GETSTATUS requests, until we hit
         // either RS2_DFU_STATE_DFU_MANIFEST_WAIT_RESET or RS2_DFU_STATE_DFU_ERROR status.
         // This command also reset the device
-        if (!wait_for_state(messenger, RS2_DFU_STATE_DFU_MANIFEST_WAIT_RESET, 20000))
-            throw std::runtime_error("Firmware manifest failed");
+        
+        if (d500_device)
+            // TODO: DFU issue - for HKR, sleep 80 seconds till finish writing to qspi flash
+            // and wake up the device with the new burned FW
+            // HKR doesn't need the next check below (RS2_DFU_STATE_DFU_MANIFEST_WAIT_RESET)
+            std::this_thread::sleep_for(std::chrono::milliseconds(80000));
```

Comment:
Maybe we can `auto devs = _ctx.query_devices();` until we see the D500 device again?
Can you check?

Reply:
Removed the sleep line
Viewer is waiting anyway till the updated device will wake again.

Reply:
But we have a timeout on the next line no?
Does it mean the user pop up will say completed and the user will see the new device after a ~1 min?
If yes it's better to keep the sleep..

Reply:
![image](https://user-images.githubusercontent.com/99127997/223560535-2c1494ff-311c-4f33-93f9-53cbaace7a80.png)

Reply:
No need for sleep.
User will see "Firmware update completed. waiting for device to reconnect..."
and no OK button will be shown till the device will be discovered again.
Then he will see the success message.